src/bms/lex/token.rs

```diff
@@ -1189,7 +1189,10 @@ impl<'a> Token<'a> {
             command if command.starts_with('#') => Self::UnknownCommand(c.next_line_entire()),
             _not_command => Self::NotACommand(c.next_line_entire()),
         };
-        Ok(token)
+
+        // Calculate the full range of this token (from command start to current cursor position)
+        let token_range = command_range.start..c.index();
+        Ok(SourceRangeMixin::new(token, token_range))
```

Comment:
```suggestion
        Ok(SourceRangeMixin::new(token, command_range))
```
I think the command's range should not include trailing whitespaces. So you don't need to extend the range.

Reply:
Oh, I got it. Its range is what consisting of all the command.